packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -0,0 +1,243 @@
+  extends TypeScriptProjectOptions {
+  /**
+   * The list of languages for which clients will be generated. A typescript client will always be generated.
+   */
+  readonly clientLanguages: ClientLanguage[];
+}
+
+/**
+ * Synthesizes a Typescript Project with an OpenAPI spec, generated clients, a CDK construct for deploying the API
+ * with API Gateway, and generated lambda handler wrappers for type-safe handling of requests.
+ *
+ * @pjid open-api-gateway-ts
+ */
+export class OpenApiGatewayTsProject extends TypeScriptProject {
+  /**
+   * A reference to the generated typescript client
+   */
+  public readonly generatedTypescriptClient: TypeScriptProject;
+
+  private readonly hasParent: boolean;
+  private readonly specDir: string = "spec";
+
+  constructor(options: OpenApiGatewayTsProjectOptions) {
+    super({
+      ...options,
+      sampleCode: false,
+      tsconfig: {
+        compilerOptions: {
+          // Root dir needs to include src and spec
+          rootDir: ".",
+          lib: ["dom", "es2019"],
+        },
+      },
+    });
+
+    // Allow spec to be imported, using both the source and spec directories as project roots.
+    this.tsconfig?.addInclude(`${this.specDir}/**/*.json`);
+    this.package.addField(
+      "main",
+      path.join(this.libdir, this.srcdir, "index.js")
+    );
+    this.package.addField(
+      "types",
+      path.join(this.libdir, this.srcdir, "index.d.ts")
+    );
+
+    // Set to private since this either uses workspaces or has file dependencies
+    this.package.addField("private", true);
+
+    // Generated project should have a dependency on this project, in order to run the generation scripts
+    this.addDeps(OPENAPI_GATEWAY_PDK_PACKAGE_NAME, "constructs", "aws-cdk-lib");
+
+    // Synthesize the openapi spec early since it's used by the generated typescript client, which is also synth'd early
+    const spec = new OpenApiSpecProject({
+      name: `${this.name}-spec`,
+      parent: this,
+      outdir: this.specDir,
+    });
+    spec.synth();
+
+    const codegenDir = "generated";
+
+    new TextFile(this, path.join(codegenDir, "README.md"), {
+      lines: [
+        "## Generated Clients",
+        "",
+        "This directory contains generated client code based on your OpenAPI Specification file (spec.yaml).",
+        "",
+        "Like other `projen` managed files, this directory should be checked in to source control, but should not be edited manually.",
+      ],
+      readonly: true,
+    });
+
+    // Parent the generated code with this project's parent for better integration with monorepos
+    this.hasParent = !!(options.parent && options.outdir);
+    const codegenDirRelativeToParent = this.hasParent
+      ? path.join(options.outdir!, codegenDir)
+      : codegenDir;
+
+    // We generate the typescript client since this project will take a dependency on it
+    this.generatedTypescriptClient = new GeneratedTypescriptClientProject({
```

Comment:
Shouldn't we only do this when clientLanguages is set to Typescript?

Reply:
Each language project will always need to generate its native client since that exports the types to make the cdk construct type-safe. I suppose I could update so we don't generate the type-safe construct if the native language isn't in the list, but I think it's a bit neater if we always do :)